icu4c/source/i18n/formattedvalue.cpp

```diff
@@ -193,6 +193,12 @@ ucfpos_close(UConstrainedFieldPosition* ptr) {
 }
 
 
+// -Wreturn-local-addr first found in https://gcc.gnu.org/onlinedocs/gcc-4.8.5/gcc/Warning-Options.html#Warning-Options
```

Comment:
A much more useful comment would be to explain _why_ it's safe to return this pointer and in what way the compiler is mistaken when it warns about that being done.

Reply:
The explaination about that is already stated inside the function previusly as 
```
// Note: this line triggers -Wreturn-local-addr, but it is safe because toTempString is
// defined to return memory owned by the ufmtval argument.
```
I do not think I need to move it here.